examples/discovery-search-app/cypress/integration/pagination/spec.ts

```diff
@@ -0,0 +1,212 @@
+const itemsPerPageOptions = ['5', '10', '20', '30', '40', '50'];
+
+describe('Pagination', () => {
+  beforeEach(() => {
+    cy.server();
+    cy.fixture('query/query.json').as('queryJSON');
+    cy.fixture('collections/collections.json').as('collectionsJSON');
+    cy.fixture('component_settings/componentSettings.json').as('componentSettingsJSON');
+    cy.fixture('query/multiPageResults.json').as('multiPageResultsJSON');
+    cy.route('GET', '**/component_settings?version=2019-01-01', '@componentSettingsJSON').as(
+      'componentSettings'
+    );
+    cy.route('POST', '**/query?version=2019-01-01', '@queryJSON').as('searchQuery');
+    cy.route('GET', '**/collections?version=2019-01-01', '@collectionsJSON').as('getCollections');
+    cy.visit('/');
+  });
+
+  describe('When the application loads', () => {
+    it('pagination component is present', () => {
+      cy.get('.bx--pagination').should('exist');
+    });
+
+    it('pagination has the correct initial settings', () => {
+      // items per page
+      cy.get('.bx--pagination__left') //TODO: change this selector to use id instead of class
+        .find('.bx--select-input')
+        .should('have.value', '5');
+      // page number
+      cy.get('.bx--pagination__right')
+        .find('.bx--select-input')
+        .should('have.value', '1');
+    });
+
+    it('items per page has the correct options available', () => {
+      itemsPerPageOptions.forEach(option => {
+        cy.get('.bx--pagination__left')
+          .find('.bx--select-input')
+          .select(option)
+          .should('have.value', option);
+      });
+    });
+  });
+
+  describe('When there are multiple pages of results', () => {
+    beforeEach(() => {
+      cy.route('POST', '**/query?version=2019-01-01', '@multiPageResultsJSON').as('multiPageQuery');
+      cy.get('.bx--search-input').type('abil{enter}');
+    });
+
+    it('lists the correct number of total pages', () => {
+      cy.get('.bx--pagination__text').should('contain', '12');
+    });
+
+    it('the previous page button is disabled', () => {
+      cy.get('.bx--pagination__button--backward').should('be.disabled');
+    });
+
+    describe('and the next page arrow is clicked', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
```

Comment:
this seems strange that the wait is before the click -> shouldn't it come after?

Reply:
If this is moved after the button click, can it also be removed from all of the following tests?

Reply:
Is this waiting on the query from https://github.com/watson-developer-cloud/discovery-components/pull/39/files#diff-5cfb6b041f06b8eb649bc618aa84fbb2R46 and the waits waiting for the request from the click below? If so, doesn't like 45 handle making sure the initial request went through? It looks like a lot of the individual tests start off with a `wait`, is there a way to restructure them to move all those `wait`s into `beforeEach`s

Reply:
That `wait` is to make sure the query went through from the `describe` this is a child of. When writing these tests, I put the wait into either the `it`s or the following `describe`s, because I usually had one `it` check that all of the request params were correct. Looking at this now, though, I bet I could move the wait up by doing something like:
```
describe('click a button' () => {
  beforeEach(() => {
    button.click();
    cy.wait('@request').as('clickedRequest');
  });

  it('the request has the correct params', () => {
    cy.get('@clickedRequest').its('whatever.param').eq('correct.value');
  )};
});
```

Reply:
I didn't think to try aliasing the request caught by `wait` before... I'll give it a shot and see how it goes.

Reply:
It worked! 🎆